runtime/drivers/clickhouse/olap.go

```diff
@@ -477,6 +540,12 @@ func (c *connection) createTable(ctx context.Context, name, sql string, outputPr
 	} else {
 		fmt.Fprintf(&create, " %s ", outputProps.Columns)
 	}
+
+	// validate incremental strategy
+	if outputProps.IncrementalStrategy == drivers.IncrementalStrategyMerge && outputProps.PartitionBy == "" {
```

Comment:
It is also possible to specify full engine directly without individual keys so I think we can just do string search for `PARTITION BY` in string returned from `tblConfig()`(https://github.com/rilldata/rill/blob/96c119ed46051c79649916826cdf43e8a1a773cd/runtime/drivers/clickhouse/model_manager.go#L104)